src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -491,4 +498,208 @@ private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOExcep
         assertEquals(1, checkouts.size());
         assertEquals(GitSCM.class, checkouts.get(0).getClass());
     }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Test proves that turning on changelog/polling and then turning it off again will remove the polling baseline
+    public void baselineResetSingleRepo() throws Exception {
+        sampleRepo.init();
+        TaskListener listener = StreamTaskListener.fromStdout();
+        //Creating the job
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        //Setting definition with changelog and polling true
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, true, true) +
+                "}\n", false));
+
+        WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+        r.assertBuildStatusSuccess(r.waitForCompletion(b1));
+        assertThat(b1.checkouts(listener).size(), equalTo(1));
+        // Checking that the polling baseline is correctly established
+        assertThat(b1.checkouts(listener).get(0).pollingBaseline, notNullValue());
+
+        //Resetting the definition to turn off changelog and polling
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                checkoutString(sampleRepo, false, false) +
+                "}\n", false));
+
+
+        WorkflowRun b2 = p.scheduleBuild2(0).waitForStart();
+
+        r.assertBuildStatusSuccess(r.waitForCompletion(b2));
+        assertThat(b2.checkouts(listener).size(), equalTo(1));
+        //proving the baseline is removed
+        assertThat(b2.checkouts(listener).get(0).pollingBaseline, nullValue());
+    }
+
+    @Test
+    @Issue("NGPIPELINE-917")
+    // Showing how only with both changelog/poll false is when it removes the checkout
+    public void baselineResetSingleRepoEdgeCases() throws Exception {
```

Comment:
This test probably needs to be changed/removed because this is "incorrect behavior"

I will do an ignore on this for now and show the correct behavior

Reply:
The "incorrect behavior" here is really only "less than ideal". It is a separate bug that should be described and fixed separately.